src/System.Private.Uri/src/System/Uri.cs

```diff
@@ -4877,6 +4877,9 @@ private unsafe char[] GetCanonicalPath(char[] dest, ref int pos, UriFormat forma
                 }
 
                 char[] dest1 = new char[dest.Length];
+
+                // We use the left-shift operator (<<) to double the length of the input to copy, since Buffer.BlockCopy
+                // takes a count of how many bytes we want to copy, whereas 'dest' is an array of chars, which are 2 bytes wide.
```

Comment:
Would the comments be necessary (or could they be simplified) if it were `end * 2` instead of `end << 1`?  I believe they should both generate the same asm.

Reply:
`end * sizeof(char)` would make more sense if so, and require less doc.

Reply:
Great idea, I'll update the PR.